packages/design-system/src/components/button.tsx

```diff
@@ -41,11 +43,16 @@ const foregrounds: Record<ButtonColor, string> = {
   neutral: theme.colors.foregroundMain,
   ghost: theme.colors.foregroundMain,
   dark: theme.colors.foregroundContrastMain,
+  gradient: theme.colors.foregroundContrastMain,
 };
 
 // CSS supports multiple gradients as backgrounds but not multiple colors
-const backgroundColors = (base: string, overlay: string) =>
-  `linear-gradient(${overlay}, ${overlay}), linear-gradient(${base}, ${base})`;
+const backgroundColors = (base: string, overlay: string) => {
+  if (base === "$colors$backgroundGradientPrimary") {
+    return `linear-gradient(${overlay}, ${overlay}), ${base}`;
+  }
```

Comment:
not sure why this was needed

Reply:
oh because base is a gradient and before it was a color

Reply:
@rpominov 
with this conditional checking that particular base the function became quite bad, it would probably make sense to rather inline that gradient and remove the function

Reply:
This solution seems ok to me, except we should replace `base === "$colors$backgroundGradientPrimary"` with `base === theme.colors.backgroundGradientPrimary`.

Logic with this solution:
 - `backgroundColors` applies two colors as a background, and does it using a gradient hack
 - but if one of the colors is a gradient itself, it should put it as is

-----------------------

Alternative solution: in `perColorStyle` do something like:

```
background: variant === "gradient" ? `linear-gradient(${overlay}, ${overlay}), ${base}` : backgroundColors(
      backgrounds[variant],
      theme.colors.backgroundButtonHover
    )
```

Not sure which one is better, both seem ok to me.

Reply:
or just do what you suggested both is kinda ok

Reply:
Hii @kof We are going with the perColorStyle solution, right?

Reply:
yeah, lets do this

Reply:
Updated and pushed the changes